src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponentResult.cs

```diff
@@ -68,83 +59,11 @@ public class ViewComponentResult : ActionResult
         public IViewEngine ViewEngine { get; set; }
 
         /// <inheritdoc />
-        public override async Task ExecuteResultAsync(ActionContext context)
-        {
-            var response = context.HttpContext.Response;
+        public override Task ExecuteResultAsync(ActionContext context)
+        {            
             var services = context.HttpContext.RequestServices;
```

Comment:
Suggest `null` checks for `context` (`ArgumentNullException`), `context.HttpContext` (`ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(...))`), and `services`.